bundle/src/main/java/com/adobe/acs/tools/csv_asset_importer/impl/CsvAssetImporterServlet.java

```diff
@@ -89,8 +89,30 @@ protected final void doPost(final SlingHttpServletRequest request, final SlingHt
         final Parameters params = new Parameters(request);
         if (params.getFile() != null) {
 
-            final long start = System.currentTimeMillis();
-            final Iterator<String[]> rows = this.getRowsFromCsv(params);
+        	Iterator<String[]> rows = null;
+			final long start = System.currentTimeMillis();
+			String inFileMimeType = params.getInFileMimeType();
+			final String tempDir = System.getProperty("java.io.tmpdir") + start;
+			if (StringUtils.equalsIgnoreCase(inFileMimeType, "application/zip")) {
+				InputStream inputStream = params.getFile();
+
+				// un-archive the zip file.
+				UnzipUtility.unzip(inputStream, tempDir);
```

Comment:
Instead of having this code in the project, what about embedding https://github.com/zeroturnaround/zt-zip ?